test/target/flag-test/target-flag.test.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+const { stat } = require('fs');
+const { resolve } = require('path');
+const { run } = require('../../utils/test-utils');
+
+const targetValues = ['web', 'webworker', 'node', 'async-node', 'node-webkit', 'electron-main', 'electron-renderer', 'electron-preload'];
+
+describe('--target flag', () => {
+    targetValues.forEach((val) => {
+        it(`should accept ${val} with --target flag`, (done) => {
+            const { stdout, stderr } = run(__dirname, ['--target', `${val}`]);
+            expect(stderr).toBeFalsy();
+            expect(stdout).toBeTruthy();
+
+            stat(resolve(__dirname, 'bin/main.js'), (err, stats) => {
+                expect(err).toBe(null);
+                expect(stats.isFile()).toBe(true);
+                done();
+            });
+        });
```

Comment:
We need to find a way to make sure `target` is used. Test on `stderr` and `stdout` is not enough

Reply:
Looking into it.

Reply:
target deals with the compilation not sure how to check if it's being used.

Reply:
/cc @rishabh3112 any suggestion?

Reply:
@snitin315 we can implement custom plugin for testing (WebpackCLITestPlugin), and output in console webpack configuration (`compiler.options`), so we can test it easy

Reply:
Nice Idea let's implement this.